Bug: #[export_endpoints] proc macro not re-exported from reinhardt facade crate

## Description

The `#[export_endpoints]` proc macro (added in PR #3695, commit 86646fb4f) is defined in `reinhardt_macros` but is not re-exported from the `reinhardt` (reinhardt-web) facade crate. This makes it unusable via the documented `#![reinhardt::export_endpoints]` inner attribute syntax in consuming crates.

## Steps to Reproduce

1. Add `#![reinhardt::export_endpoints]` to a view module file (as shown in the app template)
2. Run `cargo check`

## Expected Behavior

The `#[export_endpoints]` macro should be available as `reinhardt::export_endpoints`, since it is intended for use via the facade crate.

## Actual Behavior

```
error[E0433]: failed to resolve: could not find `export_endpoints` in `reinhardt`
 --> dashboard/src/apps/auth/views.rs:2:15
  |
2 | #![reinhardt::export_endpoints]
  |               ^^^^^^^^^^^^^^^^ could not find `export_endpoints` in `reinhardt`
```

## Analysis

In `src/lib.rs`, other proc macros are explicitly re-exported:
```rust
pub use reinhardt_macros::routes;
pub use reinhardt_macros::url_patterns;
pub use reinhardt_macros::viewset;
```

But `export_endpoints` is missing from these re-exports.

Note: The `pub use reinhardt_macros::*;` in the `macros` submodule (L283) does not make the macro accessible as `reinhardt::export_endpoints` because it's scoped under `reinhardt::macros`.

## Fix

Add to `src/lib.rs`:
```rust
pub use reinhardt_macros::export_endpoints;
```

## Environment

- reinhardt-web: commit 69c4abca (latest main)
- Discovered in reinhardt-cloud project

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-pages/macros/src/server_fn.rs
+++ b/crates/reinhardt-pages/macros/src/server_fn.rs
@@ -955,10 +955,10 @@ fn generate_server_handler(
 
 	// MSW: Generate MockableServerFn impl only when BOTH conditions are met:
 	// 1. The macro crate was compiled with `msw` feature (compile-time guard)
-	// 2. The consuming crate has `msw` feature enabled (runtime env var check)
+	// 2. The consuming crate has `msw` feature enabled (proc-macro expansion-time env var check)
 	//
-	// The compile-time guard (`cfg!`) eliminates the MSW code path entirely when
-	// the macro crate is built without `msw`, preventing any possibility of
+	// The compile-time guard (`cfg!`) avoids the MSW branch during proc-macro expansion
+	// when the macro crate is built without `msw`, preventing any possibility of
 	// env var leakage from the dependency graph. The env var check handles the
 	// case where the macro crate has `msw` but the consuming crate does not.
 	// This avoids emitting `#[cfg(feature = "msw")]` in generated code, which

--- a/crates/reinhardt-pages/macros/src/server_fn.rs
+++ b/crates/reinhardt-pages/macros/src/server_fn.rs
@@ -953,11 +953,17 @@ fn generate_server_handler(
 	// and a `use` item with the same name in the same module.
 	let marker_module_name = name.clone();
 
-	// MSW: Check at proc-macro expansion time whether the consuming crate has
-	// the "msw" feature enabled. This avoids emitting #[cfg(feature = "msw")]
-	// in the generated code, which would trigger unexpected_cfgs warnings in
-	// crates that don't declare "msw" as a known feature. (Issue #3673)
-	let msw_enabled = std::env::var("CARGO_FEATURE_MSW").is_ok();
+	// MSW: Generate MockableServerFn impl only when BOTH conditions are met:
+	// 1. The macro crate was compiled with `msw` feature (compile-time guard)
+	// 2. The consuming crate has `msw` feature enabled (runtime env var check)
+	//
+	// The compile-time guard (`cfg!`) eliminates the MSW code path entirely when
+	// the macro crate is built without `msw`, preventing any possibility of
+	// env var leakage from the dependency graph. The env var check handles the
+	// case where the macro crate has `msw` but the consuming crate does not.
+	// This avoids emitting `#[cfg(feature = "msw")]` in generated code, which
+	// would trigger unexpected_cfgs warnings in consuming crates. (Issue #3673, #3700)
+	let msw_enabled = cfg!(feature = "msw") && std::env::var("CARGO_FEATURE_MSW").is_ok();
 
 	// MSW: Extract the Ok type from Result<T, ServerFnError> for MockableServerFn::Response
 	let response_type = extract_result_ok_type(return_type);

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -294,6 +294,8 @@ pub use reinhardt_macros::routes;
 pub use reinhardt_macros::url_patterns;
 #[cfg(native)]
 pub use reinhardt_macros::viewset;
+#[cfg(native)]
+pub use reinhardt_macros::export_endpoints;
 
 // client_routes! proc macro removed: superseded by #[url_patterns(client = true)]
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -289,13 +289,13 @@ pub use reinhardt_macros::{api_view, delete, get, patch, post, put};
 
 // Re-export routes attribute macro for URL pattern registration
 #[cfg(native)]
+pub use reinhardt_macros::export_endpoints;
+#[cfg(native)]
 pub use reinhardt_macros::routes;
 #[cfg(native)]
 pub use reinhardt_macros::url_patterns;
 #[cfg(native)]
 pub use reinhardt_macros::viewset;
-#[cfg(native)]
-pub use reinhardt_macros::export_endpoints;
 
 // client_routes! proc macro removed: superseded by #[url_patterns(client = true)]
 

